etcd/server.sls

```diff
@@ -5,6 +5,31 @@ etcd_packages:
   pkg.installed:
   - names: {{ server.pkgs }}
 
+{%- if server.engine == 'kubernetes' %}
```

Comment:
Metadata is reclass-specific. That's why defaults should be also covered in states or map.jinja - to work with non-reclass users (and possibly us if we decide to drop it once).